generated/nifake/nifake/session.py

```diff
@@ -746,7 +746,8 @@ def __enter__(self):
         return self
 
     def __exit__(self, exc_type, exc_value, traceback):
-        self.close()
+        if self._interpreter._auto_close_session:
```

Comment:
We really need to move away from the "auto-close" terminology, even if (or especially if) internal only.
It collides with a different concept in our drivers, which is about closing session 1 when a user initializes session 2 to the same resource.

Can you call this `_close_on_exit` or something like that?

Reply:
Good name change suggestion, I'll do that.